### PR TITLE
Update our Actions to avoid set-output command

### LINKED ISSUE #1560: Update our Actions to avoid set-output command
### Description
A number of times in our Github Actions we use the `set-output` command to pass state from one stage to another. That is being deprecated in a number of months, and we should migrate to Github's new recommend path. 

### Proposed solution

A workflow using save-state or set-output like the following
```
- name: Save state
  run: echo "::save-state name={name}::{value}"
- name: Set output
  run: echo "::set-output name={name}::{value}"
```

should be updated to write to the new GITHUB_STATE and GITHUB_OUTPUT environment files:
```
- name: Save state
  run: echo "{name}={value}" >> $GITHUB_STATE
- name: Set output
  run: echo "{name}={value}" >> $GITHUB_OUTPUT
```

### Additional context

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

### PR DESCRIPTION
**Pull request by SaumyaBhushan**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=zC5YZpW)
Signed-off-by: SaumyaBhushan <saumya@knoldus.com>

## Summary

Update our Actions to avoid set-output command

## Output

migrate to Github's new recommend path.

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

https://github.com/buildpacks/pack/issues/1560
Resolves #1560

### CODE REVIEW

**Comment by SaumyaBhushan** [comment]
@dfreilich @natalieparellano Can you please review the pull request and approve it.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -247,7 +247,7 @@ jobs:
         id: lifecycle_version
         run: |
           LIFECYCLE_VERSION=$(./pack-linux/pack report | grep 'Default Lifecycle Version:' | grep -o '[^ ]*$')
-          echo "::set-output name=version::$LIFECYCLE_VERSION"
+          echo "version::$LIFECYCLE_VERSION" >> $GITHUB_OUTPUT
```
Comment (samj1912) [inline .github/workflows/build.yml:250]:
```suggestion
          echo "version=$LIFECYCLE_VERSION" >> $GITHUB_OUTPUT
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -257,7 +257,7 @@ jobs:
           PACK_HELP="${PACK_HELP//'%'/'%25'}"
           PACK_HELP="${PACK_HELP//$'\n'/'%0A'}"
           PACK_HELP="${PACK_HELP//$'\r'/'%0D'}"
-          echo "::set-output name=help::$PACK_HELP"
+          echo "help::$PACK_HELP" >> $GITHUB_OUTPUT
```
Comment (samj1912) [inline .github/workflows/build.yml:260]:
```suggestion
          echo "help=$PACK_HELP" >> $GITHUB_OUTPUT
```

File: .github/workflows/delivery-archlinux.yml [hunk 3]
```diff
@@ -45,8 +45,8 @@ jobs:
 
           curl -sSL "$url" -o "$fullpath"
           sha512=$(sha512sum "$fullpath" | awk '{ print $1 }')
-          echo "::set-output name=url::$url"
-          echo "::set-output name=sha512::$sha512"
+          echo "url::$url" >> $GITHUB_OUTPUT
```
Comment (samj1912) [inline .github/workflows/delivery-archlinux.yml:48]:
```suggestion
          echo "url=$url" >> $GITHUB_OUTPUT
```

File: .github/workflows/delivery-archlinux.yml [hunk 4]
```diff
@@ -45,8 +45,8 @@ jobs:
...
+          echo "sha512::$sha512" >> $GITHUB_OUTPUT
```
Comment (samj1912) [inline .github/workflows/delivery-archlinux.yml:49]:
```suggestion
          echo "sha512=$sha512" >> $GITHUB_OUTPUT
```

File: .github/workflows/delivery-archlinux.yml [hunk 5]
```diff
@@ -120,9 +120,9 @@ jobs:
         run: |
           curl -sSL ${{ steps.assets.outputs.linux_url }} -o ${{ steps.assets.outputs.linux_name }}
           sha512=$(sha512sum ${{ steps.assets.outputs.linux_name }} | cut -d ' ' -f1)
-
-          echo "::set-output name=url::${{ steps.assets.outputs.linux_url }}"
-          echo "::set-output name=sha512::$sha512"
+          
+          echo "url::${{ steps.assets.outputs.linux_url }}" >> $GITHUB_OUTPUT
+          echo "sha512::$sha512" >> $GITHUB_OUTPUT
```
Comment (samj1912) [inline .github/workflows/delivery-archlinux.yml:125]:
```suggestion
          echo "sha512=$sha512" >> $GITHUB_OUTPUT
```

File: .github/workflows/delivery-archlinux.yml [hunk 6]
```diff
@@ -120,9 +120,9 @@ jobs:
         run: |
           curl -sSL ${{ steps.assets.outputs.linux_url }} -o ${{ steps.assets.outputs.linux_name }}
           sha512=$(sha512sum ${{ steps.assets.outputs.linux_name }} | cut -d ' ' -f1)
-
-          echo "::set-output name=url::${{ steps.assets.outputs.linux_url }}"
-          echo "::set-output name=sha512::$sha512"
+          
+          echo "url::${{ steps.assets.outputs.linux_url }}" >> $GITHUB_OUTPUT
```
Comment (samj1912) [inline .github/workflows/delivery-archlinux.yml:124]:
```suggestion
          echo "url=${{ steps.assets.outputs.linux_url }}" >> $GITHUB_OUTPUT
```

File: .github/workflows/delivery-archlinux.yml [hunk 7]
```diff
@@ -120,9 +120,9 @@ jobs:
         run: |
           curl -sSL ${{ steps.assets.outputs.linux_url }} -o ${{ steps.assets.outputs.linux_name }}
           sha512=$(sha512sum ${{ steps.assets.outputs.linux_name }} | cut -d ' ' -f1)
-
-          echo "::set-output name=url::${{ steps.assets.outputs.linux_url }}"
-          echo "::set-output name=sha512::$sha512"
+
```
Comment (samj1912) [inline .github/workflows/delivery-archlinux.yml:123]:
```suggestion
```

**Review by samj1912 — CHANGES_REQUESTED** [review]

**Comment by SaumyaBhushan** [comment]
Hi @samj1912 It has been updated. please check.

**Pull request merged**